src/System.Net.WebSockets.Client/src/System/Net/WebSockets/WebSocketHandle.Unix.cs

```diff
@@ -2,108 +2,1037 @@
+                t_readLineBuffers = default(KeyValuePair<StringBuilder, byte[]>);
+                if (readLineBuffers.Key == null)
+                {
+                    readLineBuffers = new KeyValuePair<StringBuilder, byte[]>(new StringBuilder(), new byte[1]);
+                }
+
+                StringBuilder sb = readLineBuffers.Key;
+                byte[] oneByte = readLineBuffers.Value;
+                Debug.Assert(sb.Length == 0, $"Expected empty StringBuilder");
+                try
+                {
+                    while (true)
+                    {
+                        int numRead = await stream.ReadAsync(oneByte, 0, 1, cancellationToken).ConfigureAwait(false);
+                        byte b = oneByte[0];
+                        if (numRead == 0 || b == '\n') break;
+                        sb.Append((char)b);
+                    }
+
+                    if (sb.Length > 0 && sb[sb.Length - 1] == '\r')
+                    {
+                        sb.Length = sb.Length - 1;
+                    }
+
+                    return sb.ToString();
+                }
+                finally
+                {
+                    sb.Clear();
+                    t_readLineBuffers = readLineBuffers;
+                }
+            }
+
+            private async Task SendFrameAsync(ArraySegment<byte> buffer, MessageOpcode opcode, bool endOfMessage, CancellationToken cancellationToken)
+            {
+                await _sendFrameAsyncLock.WaitAsync().ConfigureAwait(false);
+                try
+                {
+                    EnsureBufferLength(ref _sendBuffer, buffer.Count + MaxMessageHeaderLength);
+
+                    int maskOffset = WriteHeader(opcode, _sendBuffer, buffer, endOfMessage);
+                    int headerLength = maskOffset + MaskLength;
+
+                    if (buffer.Count > 0)
+                    {
+                        for (int i = 0; i < buffer.Count; i++)
+                        {
+                            _sendBuffer[i + headerLength] = (byte)(buffer.Array[buffer.Offset + i] ^ _sendBuffer[maskOffset + (i % MaskLength)]);
+                        }
+                    }
+
+                    await _stream.WriteAsync(_sendBuffer, 0, headerLength + buffer.Count, cancellationToken).ConfigureAwait(false);
+                }
+                catch (ObjectDisposedException ode)
+                {
+                    throw new WebSocketException(WebSocketError.ConnectionClosedPrematurely, ode);
+                }
+                finally
+                {
+                    _sendFrameAsyncLock.Release();
+                }
+            }
+
+            private void SendKeepAliveFrameAsync()
+            {
+                Task pingTask = SendFrameAsync(new ArraySegment<byte>(Array.Empty<byte>()), MessageOpcode.Ping, true, CancellationToken.None);
+                // This is exists purely to keep the connection alive; ignore any failures.
```

Comment:
I explicitly don't want to (that's what the comment was trying to get at, but I guess I didn't explain enough).  I simply want to initiate the operation, and there's no need to block this thread waiting for it to complete. The only reason to do so would be if we wanted to propagate any failures on this thread, but I want to ignore any failures.